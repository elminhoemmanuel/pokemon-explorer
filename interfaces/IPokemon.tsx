export interface IPokemonsEach {
    name: string,
    img: string
}

export interface IPokemonsResult {
    name: string,
    url: string
}
export interface IResult {
    count: number,
    next: string | null,
    previous: string | null,
    results: IPokemonsResult[]
}

export interface IFetch{ 
    loading:boolean,
    error:string
}

export interface IBasic {
    name:string,
    url:string
}

export interface IStats{
    base_stat:number,
    effort:0,
    stat:IBasic
}
export interface IType{
    slot:number,
    type:IBasic
}
export interface IVersion{
    level_learned_at:number,
    move_learn_method:IBasic,
    version_group:IBasic
}

export interface IMove{
    move:IBasic,
    version_group_details:IVersion[]
}

export interface IPokemonDetails{ 
    species:IBasic,
    stats:IStats[],
    types:IType[],
    weight:number,
    moves:IMove[],
    [extra:string]:any

}

export interface IPagination{
    next:string | null,
    previous:string | null,
    nextPage:()=>void,
    prevPage:()=>void,
}
