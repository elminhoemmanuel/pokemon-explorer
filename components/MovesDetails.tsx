import React from 'react'
import { IMove } from 'interfaces/IPokemon'

type MovesListProps = {
    header:string,
    iterator: IMove[]
}

const MovesList = ({ header, iterator }:MovesListProps) => {
    return (
        <div>
            <p><span className="font-bold underline text-lg">{header}</span> :</p>
            <div data-testid="moves-container" className="pl-4 mb-6">
                {
                    iterator?.map(item => (
                        <p key={item.move.name} >{item.move.name}</p>
                    ))
                }
            </div>
        </div>
    )
}

export default MovesList