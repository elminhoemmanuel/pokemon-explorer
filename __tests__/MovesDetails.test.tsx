import { render, screen } from '@testing-library/react'
import MovesDetails from '../components/MovesDetails'
import { within } from '@testing-library/dom'
import { IMove } from "../interfaces/IPokemon"

const testMoves: IMove[] = [
    {
        "move": {
            "name": "razor-wind",
            "url": "https://pokeapi.co/api/v2/move/13/"
        },
        "version_group_details": [
            {
                "level_learned_at": 0,
                "move_learn_method": {
                    "name": "egg",
                    "url": "https://pokeapi.co/api/v2/move-learn-method/2/"
                },
                "version_group": {
                    "name": "gold-silver",
                    "url": "https://pokeapi.co/api/v2/version-group/3/"
                }
            },
            {
                "level_learned_at": 0,
                "move_learn_method": {
                    "name": "egg",
                    "url": "https://pokeapi.co/api/v2/move-learn-method/2/"
                },
                "version_group": {
                    "name": "crystal",
                    "url": "https://pokeapi.co/api/v2/version-group/4/"
                }
            }
        ]
    },

]


test("On render, check that moves is displayed correctly", () => {
    const { container } = render(
        <MovesDetails header="Moves" iterator={testMoves} />
    );
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot()
})