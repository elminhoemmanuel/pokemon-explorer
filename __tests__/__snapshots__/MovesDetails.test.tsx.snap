// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On render, check that moves is displayed correctly 1`] = `
<div>
  <p>
    <span
      class="font-bold underline text-lg"
    >
      Moves
    </span>
     :
  </p>
  <div
    class="pl-4 mb-6"
    data-testid="moves-container"
  >
    <p>
      razor-wind
    </p>
  </div>
</div>
`;

exports[`On render, check that stats is displayed correctly 1`] = `
<div>
  <p>
    <span
      class="font-bold underline text-lg"
    >
      Stats
    </span>
     :
  </p>
  <div
    class="pl-4 mb-6"
    data-testid="stats-container"
  >
    <p>
      hp
       - 
      45
    </p>
    <p>
      attack
       - 
      49
    </p>
  </div>
</div>
`;
